server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachment.java

```
@@ -0,0 +1,107 @@
+ *
+ * <br />
+ * It does not takes any parameter
+ *
+ * Sample configuration:
+ * <p/>
+ * <pre><code>
+ * &lt;mailet match="All" class="TextCalendarBodyToAttachment"/&gt;
+ * </code></pre>
+ *
+ */
+public class TextCalendarBodyToAttachment extends GenericMailet {
+    private static final String TEXT_CALENDAR_TYPE = "text/calendar";
+    public static final String HEADER_PREFIX_CONTENT = "Content-";
+
+    @Override
+    public String getMailetInfo() {
+        return "Moves body part of content type text/calendar to attachment";
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        MimeMessage mimeMessage = mail.getMessage();
+        if (mimeMessage.isMimeType(TEXT_CALENDAR_TYPE)) {
+            processTextBodyAsAttachment(mimeMessage);
+        }
+    }
+
+    @VisibleForTesting
+    void processTextBodyAsAttachment(MimeMessage mimeMessage) throws MessagingException {
+        Multipart multipart = new MimeMultipart();
+        multipart.addBodyPart(getMimeBodyPart(mimeMessage, mimeMessage.getRawInputStream()));
+
+        mimeMessage.setContent(multipart);
+        mimeMessage.saveChanges();
+    }
+
+    private MimeBodyPart getMimeBodyPart(MimeMessage mimeMessage, InputStream mimeContent) throws MessagingException {
+        MimeBodyPart fileBody = new MimeBodyPart(mimeContent);
+
+        fileBody = moveHeaderContentFromMimeMessageToBodyPart(mimeMessage, fileBody);
+
+        fileBody.setDisposition(Part.ATTACHMENT);
+        return fileBody;
+    }
+
+    private MimeBodyPart moveHeaderContentFromMimeMessageToBodyPart(MimeMessage mimeMessage, MimeBodyPart fileBody) throws MessagingException {
+        List<Header> contentHeaders = Collections.list((Enumeration<Header>) mimeMessage.getAllHeaders())
+            .stream()
+            .filter(header -> header.getName().startsWith(HEADER_PREFIX_CONTENT))
+            .collect(Guavate.toImmutableList());
```

Just a detail, but you don't need to `collect`; you can chain with a `forEach`.

---

As MessageException is not a runtime exception, it makes error exception painfull.

With Throwing lambda you can:
 - Ignore the exception (not cool)
 - Or throw it as a Runtime exception (not cool either)

And I do believe exception unboxing is overkill. I would advocate *keep it like this*.

---

And that's why I'm mentioned that do not use "forEach"